Stop using IsDeprecatedTimerSmartPointerException in WebKit/NetworkProcess

cdumez opened the pull request:
#### e5251715c351a5e775065a94321579b97b664f02
<pre>
Stop using IsDeprecatedTimerSmartPointerException in WebKit/NetworkProcess
<a href="https://bugs.webkit.org/show_bug.cgi?id=282885">https://bugs.webkit.org/show_bug.cgi?id=282885</a>

Reviewed by Geoffrey Garen.

* Source/WebKit/NetworkProcess/NetworkSession.cpp:
(WebKit::NetworkSession::CachedNetworkResourceLoader::create):
(WebKit::NetworkSession::addLoaderAwaitingWebProcessTransfer):
* Source/WebKit/NetworkProcess/NetworkSession.h:
* Source/WebKit/NetworkProcess/cache/PrefetchCache.h:
(WebKit::PrefetchCache::Entry::releaseBuffer): Deleted.
* Source/WebKit/NetworkProcess/glib/DNSCache.cpp:
(WebKit::DNSCache::create):
* Source/WebKit/NetworkProcess/glib/DNSCache.h:
* Source/WebKit/NetworkProcess/glib/WebKitCachedResolver.cpp:

Canonical link: <a href="https://commits.webkit.org/286416@main">https://commits.webkit.org/286416@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9d5a169b045e4c875800307e3d84da7f4d0940f6

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/75935 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/54964 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/28832 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/80432 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/27201 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/64106 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/3260 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/59539 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/17699 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/79002 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/49416 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/65210 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/39899 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/46816 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/25528 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/67931 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/23031 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/81896 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/3304 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/2091 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/67767 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/3458 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/65178 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/67076 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16711 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/11023 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/9144 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/3254 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/6060 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/3275 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/4213 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/3282 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

geoffreygaren commented on Source/WebKit/NetworkProcess/cache/PrefetchCache.h:
```diff
@@ -30,25 +30,18 @@
-namespace WebKit {
-
-class PrefetchCache {
+class PrefetchCache final : public CanMakeCheckedPtr<PrefetchCache> {
```
Will this create new tool failures? (I don’t see any CheckedPtr additions in this patch.)

cdumez replied:
[mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/6034) seems fine. Maybe the CheckedPtr are still disabled though?

cdumez replied:
I think the checks must be disabled still. I have found a few places that needed a CheckedRef indeed. Fixing them.

rniwa replied:
We haven't enabled checked ptr checkers yet.

cdumez committed:
Stop using IsDeprecatedTimerSmartPointerException in WebKit/NetworkProcess
https://bugs.webkit.org/show_bug.cgi?id=282885

Reviewed by Geoffrey Garen.

* Source/WebKit/NetworkProcess/NetworkSession.cpp:
(WebKit::NetworkSession::CachedNetworkResourceLoader::create):
(WebKit::NetworkSession::addLoaderAwaitingWebProcessTransfer):
* Source/WebKit/NetworkProcess/NetworkSession.h:
* Source/WebKit/NetworkProcess/cache/PrefetchCache.h:
(WebKit::PrefetchCache::Entry::releaseBuffer): Deleted.
* Source/WebKit/NetworkProcess/glib/DNSCache.cpp:
(WebKit::DNSCache::create):
* Source/WebKit/NetworkProcess/glib/DNSCache.h:
* Source/WebKit/NetworkProcess/glib/WebKitCachedResolver.cpp:

Canonical link: https://commits.webkit.org/286416@main
